develop/toolkit/api-libraries/smart-contracts/ether-js.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Ethers.Js
+description: Learn how to interact with the Asset Hub chain using Ethers.js, compiling and deploying Solidity contracts, and interacting with deployed smart contracts.
+---
+
+# Ethers.Js
+
+## Introduction
+
+Ethers.js is a lightweight library that enables interaction with Ethereum Virtual Machine (EVM)-compatible blockchains through JavaScript. This article demonstrates how to use Ethers.js to interact and deploy smart contracts to the Asset Hub.
+
+## Set Up the Project
+
+To start working with Ethers.js, begin by initializing your project:
+
+```bash
+npm init -y
+```
+
+## Install Dependencies
+
+Next, install the Ethers.js library:
+
+```bash
+npm install ethers
+```
+
+## Set Up the Ethers.js Provider
+
+To interact with the Asset Hub, you'll need to set up an Ethers.js provider. This provider connects to a blockchain node, allowing you to query blockchain data and interact with smart contracts. Here's how to configure it:
+
+```js
+--8<-- 'code/develop/toolkit/api-libraries/smart-contracts/ether-js/connectToProvider.js'
+```
+
+!!! note
+    Replace `INSERT_RPC_URL`, `INSERT_CHAIN_ID`, and `INSERT_CHAIN_NAME` with the appropriate values. For example, to connect to Westend Asset Hub, you can use the following parameters:
+
+    ```js
+    const PROVIDER_RPC = {
+        rpc: 'https://westend-asset-hub-eth-rpc.polkadot.io',
+        chainId: 420420421,
+        name: 'westend-asset-hub'
+    };
+    ```
+
+With the [`Provider`](https://docs.ethers.org/v6/api/providers/#Provider){target=\_blank} set up, you can start querying the blockchain. For instance, to fetch the latest block number:
+
+??? code "Fetch Last Block code"
+
+    ```js
+    --8<-- 'code/develop/toolkit/api-libraries/smart-contracts/ether-js/fetchLastBlock.js'
+    ```
+
+## Compile Contracts
+
+To deploy smart contracts to Asset Hub, you need to compile them into `polkavm` bytecode. Use the [`@parity/revive`](https://www.npmjs.com/package/@parity/revive){target=\_blank} library, which compiles Solidity code for use on Substrate-based chains.
```

Comment:
Might be worth linking something about what PolkaVM is eventually, not sure if we have a good page here, if not we can make a follow up issue as I don't think it should block this PR.